xscen/extract.py

```diff
@@ -1078,3 +1084,154 @@ def _subset_file_coverage(
+    tas_csv: str = None,
+    to_level: str = "warminglevel-{wl}vs{period0}-{period1}",
+    wl_dim: str = "+{wl}Cvs{period0}-{period1}",
+):
+    """
+    Subsets the input dataset with only the window of time over which the requested level of
+    global warming is first reached, using the IPCC Atlas method.
+
+
+    Parameters
+    ----------
+    ds: xr.Dataset
+      Input dataset.
+      The dataset should include attributes to help recognize it and find its
+      warming levels - 'cat:mip_era', 'cat:experiment', 'cat:member', and either
+      'cat:source' for global models or 'cat:driving_model' for regional models.
+    wl: float
+      Warming level.
+      eg. 2 for a global warming level of +2 degree Celsius above the mean temperature of the `tas_baseline_period`.
+    window: int
+      Size of the rolling window in years over which to compute the warming level.
+    tas_baseline_period: list
+      Base period. The warming is calculated with respect to this period. The default is ["1850", "1900"].
+    ignore_member: bool
+      Whether to ignore the member when searching for the model run in tas_csv.
+    tas_csv: str
+      Path to a csv of annual global mean temperature with a row for each year and a column for each dataset.
+      If None, it will default to data/IPCC_annual_global_tas.csv which was built from
+      the IPCC atlas data from  Iturbide et al., 2020 (https://doi.org/10.5194/essd-12-2959-2020)
+      and extra data from pilot models of MRCC5 and ClimEx.
+    to_level:
+      The processing level to assign to the output.
+      Use "{wl}", "{period0}" and "{period1}" in the string to dynamically include
+      `wl`, 'tas_baseline_period[0]' and 'tas_baseline_period[1]'.
+    wl_dim: str
+      The value to use to fill the new `warminglevel` dimension.
+      Use "{wl}", "{period0}" and "{period1}" in the string to dynamically include
+      `wl`, 'tas_baseline_period[0]' and 'tas_baseline_period[1]'.
+      If None, no new dimensions will be added.
+
+    Returns
+    -------
+    xr.Dataset
+        Warming level dataset.
+
+    """
+
+    if tas_baseline_period is None:
+        tas_baseline_period = ["1850", "1900"]
+
+    if tas_csv is None:
+        tas_csv = Path(__file__).parent / "data/IPCC_annual_global_tas.csv"
+
+    # get info on ds
+    id_ds = ds.attrs["cat:id"]
+    source_ds = (
+        ds.attrs["cat:source"]
+        if pd.isna(ds.attrs.get("cat:driving_model", None))
+        else ds.attrs["cat:driving_model"]
+    )
+    exp_ds = ds.attrs["cat:experiment"]
+    member_ds = ds.attrs["cat:member"]
+    mip_era_ds = ds.attrs["cat:mip_era"]
+
+    info_ds = (
+        f"{mip_era_ds}_{source_ds}_{exp_ds}_.*"
+        if ignore_member
+        else f"{mip_era_ds}_{source_ds}_{exp_ds}_{member_ds}"
+    )
+
+    # open csv
+    annual_tas = pd.read_csv(tas_csv, index_col="year")
+
+    # choose colum based in ds cat attrs
+    right_column = annual_tas.filter(regex=info_ds, axis=1)
```

Comment:
```suggestion
    right_column = annual_tas.filter(regex=re.compile(info_ds, re.IGNORECASE), axis=1)
```
case insensitive version